test/unit_test/model/test_onnx_model.py

```diff
@@ -223,7 +222,7 @@ def test_model_prepare_session_with_unsupported_eps(get_available_providers_mock
     execution_providers = ["TensorrtExecutionProvider", "CUDAExecutionProvider"]
     # provider_options = [{}, {"device_id": "1"}]
     with pytest.raises(
-        OliveEvaluationError,
+        RuntimeError,
```

Comment:
what's the reason that change the exception type? the OliveEvaluationError will be ignored by engine but the RuntimeError will not

Reply:
The error is raised by the `ort_inference` util module so it doesn't have access to this error class. 

I could catch this error and reraise with the correct error during `model.create_session` if that's better?

Reply:
yes, for onnx model, we need raise OliveEvaluationError. the only thing is how could know the RuntimeError is the EP fallback issue instead of other RuntimeError? do we need create a customized exception in that file?

Reply:
That sounds like a workable idea to me. I can do that if you are okay with it too.

Reply:
done!